Fix logic in check_exptime

stsci-hack opened the pull request:
The logic in check_exptime was setup to remove filenames from the input list.  However, the input list it now gets is always a list of HDUList objects, not filenames.  This code was therefore changed to try to remove the HDUList objects for exposures with exptime==0 instead of the filename.  This problem was identified in HSTDP 2019.2 regression testing, with the fix in this PR being tested against the dataset reported by HSTDP as well as another ASN where exptime was valid.

nden commented:
Could you add a quick test? it doesn't have to use files, may be an HDUList object in memory.

stsci-hack commented:
A new test has been added, and the test for check_NGOODPIX was also updated to reflect the same change in logic. This process also uncovered a (hopefully, final) bug with filename handling that was also fixed.
